pkg/kubectl/cmd/set/set_selector.go

```diff
@@ -0,0 +1,273 @@
+		return fmt.Errorf("one or more resources must be specified as <resource> <name> or <resource>/<name>")
+	}
+	return nil
+}
+
+// copyMatchLabels copy match labels from src to dst
+func copyMatchLabels(dst *map[string]string, src *map[string]string, overwrite bool) {
+	if overwrite {
+		*dst = make(map[string]string)
+	}
+	for label, value := range *src {
+		_, exists := (*dst)[label]
+		if overwrite || !exists {
+			(*dst)[label] = value
+		}
+	}
+}
+
+// copySelector copy match labels and match expressions from src to dst
+func copySelector(dst *unversioned.LabelSelector, src *unversioned.LabelSelector, overwrite bool) {
+	copyMatchLabels(&dst.MatchLabels, &src.MatchLabels, overwrite)
+	if overwrite || dst.MatchExpressions == nil {
+		dst.MatchExpressions = make([]unversioned.LabelSelectorRequirement, len(src.MatchExpressions))
+	}
+	if overwrite {
+		copy(dst.MatchExpressions, src.MatchExpressions)
+		return
+	}
+	for _, srcExp := range src.MatchExpressions {
+		found := false
+		for _, dstExp := range dst.MatchExpressions {
+			if srcExp.Key == dstExp.Key {
+				found = true
+				break
+			}
+		}
+		if !found {
+			dst.MatchExpressions = append(dst.MatchExpressions, srcExp)
+		}
+	}
+}
+
+// Run the command.
+func (o *SelectorOptions) Run() error {
+	allErrs := []error{}
+
+	patches := CalculatePatches(o.Infos, o.Encoder, func(info *resource.Info) (bool, error) {
+		err := o.UpdateSelectorForObject(info.Object, func(map_selector *map[string]string, selector *unversioned.LabelSelector) error {
+			transformed := false
+			if map_selector != nil {
+				if len(o.Selector.MatchExpressions) == 0 {
+					transformed = true
+					copyMatchLabels(map_selector, &o.Selector.MatchLabels, o.Overwrite)
+				} else {
+					allErrs = append(allErrs, fmt.Errorf("error: MatchExpression %v not supported on this object", o.Selector.MatchExpressions))
+				}
+			}
+			if selector != nil {
+				copySelector(selector, o.Selector, o.Overwrite)
+				transformed = true
+			}
+			if !transformed {
+				allErrs = append(allErrs, fmt.Errorf("error: selector not updated %v", o.Selector))
+			}
+
+			return nil
+		})
+		return true, err
+	})
+
+	for _, patch := range patches {
+		info := patch.Info
+		if patch.Err != nil {
+			allErrs = append(allErrs, fmt.Errorf("error: %s/%s %v\n", info.Mapping.Resource, info.Name, patch.Err))
+			continue
+		}
+		// no changes
+		if string(patch.Patch) == "{}" || len(patch.Patch) == 0 {
+			continue
+		}
+
+		if o.Local || o.DryRun {
+			fmt.Fprintln(o.Out, "running in local/dryRun mode...")
+			return o.PrintObject(o.Cmd, o.Mapper, info.Object, o.Out)
+		}
+
+		// patch the change
+		obj, err := resource.NewHelper(info.Client, info.Mapping).Patch(info.Namespace, info.Name, api.StrategicMergePatchType, patch.Patch)
+		if err != nil {
+			allErrs = append(allErrs, fmt.Errorf("failed to patch selector update to template: %v\n", err))
+			continue
+		}
+		info.Refresh(obj, true)
+
+		// record this change (for rollout history)
```

Comment:
@kargakis to be honest I copied this little chunk from set_image.go, so no smart reason :-O